frontend/dockerfile/dockerfile2llb/convert.go

```diff
@@ -1573,6 +1579,8 @@ func dispatchArg(d *dispatchState, c *instructions.ArgCommand, metaArgs []instru
 		if buildArg.Value != nil {
 			if _, ok := nonEnvArgs[buildArg.Key]; !ok {
 				d.state = d.state.AddEnv(buildArg.Key, *buildArg.Value)
+			} else {
+				d.state = d.state.AddEnv(buildArg.Key, "")
```

Comment:
This looks like a behavior change? `ARG FOO` should not make `FOO` show up for `RUN` commands, or I think this is the current behavior. This could have a integration test if it is not detected atm.

Reply:
> ARG FOO should not make FOO show up for RUN commands, or I think this is the current behavior. This could have a integration test if it is not detected atm.

That's not what the change seems to be doing as it looks for `nonEnvArgs`, but still the previous code looks more correct to me.